### ISSUE TITLE
Tracking of projects that fail to run coverage job

### ISSUE THREAD

**Issue by Dor1s** [google/oss-fuzz#1743]
- [x] **bad_example**
Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/bad_example/targets.list.address".
as the target is not being run on CF
- [x] **chakra**
The same root cause: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/chakra/targets.list.address".
- [x] **dlplibs**
the disk is full, cannot unpack the corpus:
```
I  Step #4: /corpus/mswksfuzzer/5baeffd6604290d2cbea9a2112181f6c7b91ea0c:  write error (disk full?).  Continue? (y/n/^C)  
I  Step #4: warning:  /corpus/mswksfuzzer/5baeffd6604290d2cbea9a2112181f6c7b91ea0c is probably truncated 
I  Step #4: checkdir:  cannot create extraction directory: /corpus/mswrdfuzzer 
I  Step #4:            No space left on device 
I  Step #4: checkdir:  cannot create extraction directory: /corpus/multiplanfuzzer 
I  Step #4:            No space left on device 
```
- [x] **example**
download_corpus failed, target is disabled on CF
- [x] **firefox**
the disk is full, cannot finish compilation
```
I  Step #2: + find media/webrtc/trunk/webrtc/test/fuzzers/corpora/stun-corpus -type f -exec zip -qju /workspace/out/profile/StunParser_seed_corpus.zip '{}' ';' 
I  Step #2: + cp media/webrtc/trunk/webrtc/test/fuzzers/corpora/stun.tokens /workspace/out/profile/StunParser.dict 
I  Step #2: cp: error writing '/workspace/out/profile/work/obj-fuzz/_tests/testing/mochitest/chrome/toolkit/mozapps/update/tests/data/updater': No space left on device 
I  Step #2: cp: cannot create directory '/workspace/out/profile/work/obj-fuzz/_tests/testing/mochitest/chrome/toolkit/components': No space left on device 
``` 
- [x] **fuzzing-puzzles**
The project is disabled: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/fuzzing-puzzles/targets.list.address".
- [x] **glib**
Compilation failure. Weird, as I've added `-ldl` flag. Looks like the project does linking using `clang`, not `clang++`:
```
I  Step #2: [773/955] Linking target gio/gsettings. 
I  Step #2: FAILED: gio/gsettings  
I  Step #2: clang  -o gio/gsettings 'gio/gio@@gsettings@exe/gsettings-tool.c.o' -Wl,--no-undefined -Wl,--as-needed -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fprofile-instr-generate -fcoverage-mapping -pthread -Wl,-ldl -Wno-unused-command-line-argument -Wl,--start-group gio/libgio-2.0.a gio/xdgmime/libxdgmime.a gio/inotify/libinotify.a glib/libglib-2.0.a glib/libcharset/libcharset.a glib/pcre/libpcre.a gobject/libgobject-2.0.a subprojects/libffi/src/libffi.a subprojects/zlib-1.2.11/libz.a gmodule/libgmodule-2.0.a -lresolv -Wl,--end-group   
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_self': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:125: undefined reference to `dlopen' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_open': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:98: undefined reference to `dlopen' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_close': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:150: undefined reference to `dlclose' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `_g_module_symbol': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:163: undefined reference to `dlsym' 
I  Step #2: gmodule/libgmodule-2.0.a(gmodule.c.o): In function `fetch_dlerror': 
I  Step #2: /work/meson/../../src/glib/gmodule/gmodule-dl.c:80: undefined reference to `dlerror' 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
```
- [x] **jsc**
The project is disabled: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/jsc/targets.list.address".
- [x] **libchewing**
The target doesn't have corpus backups as it's disabled: https://github.com/google/oss-fuzz/blob/master/projects/libchewing/project.yaml#L2
- [x] **libprotobuf-mutator**
Compilation failure without a reasonable error message:
```
I  Step #2: [208/209] Building CXX object CMakeFiles/protoc.dir/src/libprotobuf-mutator/build/external.protobuf/src/external.protobuf/src/google/protobuf/compiler/main.cc.o 
I  Step #2: [209/209] Linking CXX executable protoc 
I  Step #2: ninja: build stopped: subcommand failed. 
I  Finished Step #2 
I  ERROR 
I  ERROR: build step 2 "gcr.io/oss-fuzz/libprotobuf-mutator" failed: exit status 1 
```
Regular build has the same problem though: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10003
- [x] **libpsl**
Compilation failure:
```
I  Step #2:   CCLD     psl 
I  Step #2: /usr/bin/ld: /src/deps/lib/libicuuc.a(putil.ao): undefined reference to symbol 'floor@@GLIBC_2.2.5' 
I  Step #2: //lib/x86_64-linux-gnu/libm.so.6: error adding symbols: DSO missing from command line 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: make[2]: *** [psl] Error 1 
I  Step #2: Makefile:468: recipe for target 'psl' failed 
```
- [x] **libssh**
Compilation failure, the same as the regular build has.
```
I  Step #2: CMake Error at cmake/Modules/CheckCCompilerFlagSSP.cmake:19 (include_guard): 
I  Step #2:   Unknown CMake command "include_guard". 
I  Step #2: Call Stack (most recent call first): 
I  Step #2:   CompilerChecks.cmake:2 (include) 
I  Step #2:   CMakeLists.txt:31 (include) 
I  Step #2:  
I  Step #2:  
I  Step #2: -- Configuring incomplete, errors occurred! 
I  Step #2: See also "/src/libssh/build/CMakeFiles/CMakeOutput.log". 
```
- [x] **mercurial**
Corpus download failed, probably because https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=9694
xdiff_fuzzer seems to be crazy slow so corpus pruning never finishes since early July
- [x] **nss**
No fuzz targets (weird): Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/nss/targets.list.address".
- [x] **openvswitch**
Build failure:
```
I  Step #2: + clang++ -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fprofile-instr-generate -fcoverage-mapping -pthread -Wl,-ldl -Wno-unused-command-line-argument -stdlib=libc++ /workspace/out/profile/flow_extract_target.o ./lib/.libs/libopenvswitch.a -lz -lssl -lcrypto -latomic -lFuzzingEngine -o /workspace/out/profile/flow_extract_fuzzer 
I  Step #2: ./lib/.libs/libopenvswitch.a(async-append-aio.o): In function `async_append_write': 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:144: undefined reference to `aio_write' 
I  Step #2: ./lib/.libs/libopenvswitch.a(async-append-aio.o): In function `async_append_wait': 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:99: undefined reference to `aio_error' 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:106: undefined reference to `aio_suspend' 
I  Step #2: /src/openvswitch/lib/async-append-aio.c:108: undefined reference to `aio_return' 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 

```
- [x] **postgis**
Corpus download failed. Missing corpus backup for postgis_wkb_import_fuzzer.
- [x] **proj4**
Build failure (messing up with `atexit` which is used for dumping coverage):
```
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: /usr/bin/ld: .libs/basic_test: hidden symbol `atexit' in /usr/lib/x86_64-linux-gnu/libc_nonshared.a(atexit.oS) is referenced by DSO 
I  Step #2: /usr/bin/ld: final link failed: Bad value 
I  Step #2: make[2]: *** [proj_errno_string_test] Error 1 
I  Step #2: Makefile:424: recipe for target 'proj_errno_string_test' failed 
I  Step #2: clang-7: error: linker command failed with exit code 1 (use -v to see invocation) 
I  Step #2: make[2]: *** [basic_test] Error 1 
I  Step #2: Makefile:416: recipe for target 'basic_test' failed 
I  Step #2: Makefile:369: recipe for target 'all-recursive' failed 
I  Step #2: make[1]: *** [all-recursive] Error 1 
I  Step #2: make: *** [all-recursive] Error 1 
I  Step #2: Makefile:448: recipe for target 'all-recursive' failed 

```
- [x] **spidermonkey**
No fuzz targets, WAI: Failed to get list of targets from "https://storage.googleapis.com/clusterfuzz-builds/spidermonkey/targets.list.address".
- [x] **strongswan**
Build failure. Not sure what's going on, by adding `-lm` likely will solve the problem:

```
I  Step #2: rm -f libFuzzerLocal.a 
I  Step #2: ar cru libFuzzerLocal.a libFuzzerLocal.o ../src/libstrongswan/libstrongswan.la 
I  Step #2: ar: `u' modifier ignored since `D' is the default (see `U') 
I  Step #2: ranlib libFuzzerLocal.a 
I  Step #2: /usr/lib/libFuzzingEngine.a(FuzzerLoop.o): In function `ceil': 
I  Step #2: /usr/local/bin/../include/c++/v1/math.h:834: undefined reference to `ceilf' 
I  Step #2: /usr/local/bin/../include/c++/v1/math.h:834: undefined reference to `ceilf' 
```

**Comment by Dor1s** [comment]
* mercurial: I've manually created `latest.zip` using the most recent corpus backup available
* postgis: looks like there is a new fuzz target added recently, I've created a dummy `latest.zip` but likely it will be overwritten soon, which is great

**Comment by Dor1s** [comment]
Once #1745 lands, probably the last thing I should fix on the OSS-Fuzz side is the issue with disk space.

@oliverchang dlplibs and firefox coverage builds do not fit the disk. I guess I can just go ahead and add `diskSizeGb` with a value greater than the default 110GB to `GCB_OPTIONS`?

**Comment by oliverchang** [comment]
> @oliverchang dlplibs and firefox coverage builds do not fit the disk. I guess I can just go ahead and add diskSizeGb with a value greater than the default 110GB to GCB_OPTIONS?

Sounds good!

**Comment by pdknsk** [comment]
Note that `firefox` should weigh less once it can be build without `ASAN` for mode _profile_. That's currently work in progress. On `glib`, I opened a bug report with the maintainers.

**Comment by Dor1s** [comment]
The issue got accidentally closed by one of the commits. Re-opening. I still need to check builds once again tomorrow and probably file separate issues to the corresponding projects that have failures.

**Comment by Dor1s** [comment]
> @oliverchang I made it 150GB for now.

That worked, now firefox seems to be failing when running `llvm-cov`: https://github.com/google/oss-fuzz/blob/master/infra/base-images/base-runner/coverage#L136 plus some warnings during merge steps

and glib has a linker failure: undefined `dlopen` and other symbols even though `-ldl` flag is being passed. Looks like they are using `clang` instead of `clang++` for linking.

**Comment by pdknsk** [comment]
Do you have logs for `firefox`? I also had warnings during merge, but the report was still good.

**Comment by Dor1s** [comment]
> Do you have logs for firefox? I also had warnings during merge, but the report was still good.

Yes, you can see the logs here now: https://oss-fuzz-build-logs.storage.googleapis.com/index.html

**Comment by Dor1s** [comment]
And the following line means that `llvm-cov` execution failed, i.e. we haven't generated the report in HTML, so the post processing errors out as well:

```
Step #4: [2018-08-26 14:54:01,333 ERROR] Default coverage report dir does not exist: /workspace/out/profile/report/coverage.
```

**Comment by Dor1s** [comment]
Bumped the disk size of 200 as grpc is still failing. Also added copying of /usr/include (needed for wireshark and irssi).

**Comment by Dor1s** [comment]
Thanks @pdknsk for the investigation, sounds good!

**Comment by pdknsk** [comment]
> Yes, you can see the logs here now: https://oss-fuzz-build-logs.storage.googleapis.com/index.html

There are no apparent errors in the [latest logs](https://oss-fuzz-build-logs.storage.googleapis.com/log-507ca118-f451-4f4a-a6a1-e4d8a457e6b6.txt). It just quits with this message.
```
ERROR: build step 3 "gcr.io/oss-fuzz-base/base-runner" failed: exit status 8
```

**Comment by Dor1s** [comment]
Step #3 should be corpus backup downloading. I guess we didn't have corpus backup at that moment, but I see that we do have it now, so let's wait a bit :)

**Comment by Dor1s** [comment]
The job which is running right now has passed that step, but failed with the same issue as it did in the past. Build status page should get updated soon and you'll see that log.

**Comment by pdknsk** [comment]
I get the same error locally.
```
$ python infra/helper.py profile firefox
...
FileNotFoundError: [Errno 2] No such file or directory: '/out/report/coverage'
Failed to generate clang code coverage report.
```
It does work when I pass additional objects.
```
$ python infra/helper.py profile firefox -- -object firefox/firefox -object firefox/libxul.so
```
I don't know why it fails without it. So now with https://github.com/google/oss-fuzz/pull/1774 fixed this should work. I'll add the flags with some other related changes soon.

The final report is 1873MB. Coverage for the whole project is obviously quite low.

![1](https://user-images.githubusercontent.com/1097609/44897586-79ccc880-acfc-11e8-8fad-d1ded8b3816e.png)

I noticed that during _profile_ runs the targets seem to ignore `options`, as `ContentParentIPC` writes a 50MB log, which probably slows it down.

**Comment by pdknsk** [comment]
And `glib` works now.

**Comment by Dor1s** [comment]
> I noticed that during profile runs the targets seem to ignore options, as ContentParentIPC writes a 50MB log, which probably slows it down.

That's a great point. I think I can enable `-close_fd_mask=3` for all fuzz targets when generating coverage report.

**Comment by Dor1s** [comment]
Landed as https://github.com/google/oss-fuzz/commit/72b82ee08ffd39e284a7271bb556ba0e5b7b3e39

Will keep an eye on the stats in a day or so.

**Comment by evverx** [comment]
I've just received a letter from OSS-Fuzz about a coverage build failure described in https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10239. Looking at the bottom of [the build log](https://oss-fuzz-build-logs.storage.googleapis.com/log-b84f0869-3c4f-437c-bb53-63bf7ed6875f.txt), it seems to me that it has nothing to do with our fuzzers:
```sh
Step #3: 6480b1022003: Download complete
Step #3: eac708e4a0ed: Download complete
Step #3: eac708e4a0ed: Pull complete
Step #3: 2d93fe48b978: Pull complete
Step #3: ff8f20fa0479: Pull complete
Step #3: 6480b1022003: Pull complete
Step #3: Digest: sha256:44208283859ce8f023f39f36ff55a19c41e69bef26a8ee851d1430fec4369b8f
Step #3: Status: Downloaded newer image for gcr.io/oss-fuzz-base/base-runner:latest
Finished Step #3
ERROR
ERROR: build step 3 "gcr.io/oss-fuzz-base/base-runner" failed: exit status 8
```

**Comment by Dor1s** [comment]
@evverx, step #3 is corpus backup downloading. We simply didn't have corpus backup for that new target you've recently added. Now the build is green.

However, we should not report issues like this. Thanks for bringing this up, I'll fix it as #1798.

**Comment by Dor1s** [comment]
> firefox works now.

Awesome!

Regarding `target.c`, as long as the fuzz targets produce corrupted dumps (see `Function basic block count change` errors below), the report might be considered "undefined", because those errors might simply interrupt merge process and do not let it to process all the counters.

```
I  Starting Step #4 
I  Step #4: Already have image (with digest): gcr.io/oss-fuzz-base/base-runner 
I  Step #4: Running ./SdpParser 
I  Step #4: Running ./ContentParentIPC 
I  Step #4: Running ./ContentSecurityPolicyParser 
I  Step #4: Running ./StunParser 
I  Step #4: /workspace/out/profile/dumps/./StunParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE7emplaceIJRDiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: Make sure that all profile data to be merged is generated from the same binary. 
I  Step #4: /workspace/out/profile/dumps/./StunParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./StunParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiEdeEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./StunParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE7emplaceIJRiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./StunParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./StunParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: [2018-09-25 14:39:48,156 INFO] Finding shared libraries for targets (if any). 
I  Step #4: [2018-09-25 14:39:48,164 INFO] Finished finding shared libraries for targets. 
I  Step #4: Error occured while running ./ContentParentIPC: 
I  Step #4: Running Fuzzer tests... 
I  Step #4: INFO: Seed: 314399180 
I  Step #4: INFO: Loaded 1 modules   (1383288 inline 8-bit counters): 1383288 [0x7fc388493388, 0x7fc3885e4f00),  
I  Step #4: INFO: Loaded 1 PC tables (1383288 PCs): 1383288 [0x7fc3885e4f00,0x7fc389b00680),  
I  Step #4: INFO:    14638 files found in /corpus/./ContentParentIPC 
I  Step #4: INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 543265 bytes 
I  Step #4: INFO: seed corpus: files: 14638 min: 1b max: 543265b total: 10799963b rss: 262Mb 
I  Step #4: #1024	pulse  cov: 1003 ft: 1109 corp: 570/17Kb lim: 4 exec/s: 512 rss: 263Mb 
I  Step #4: #2048	pulse  cov: 3838 ft: 4391 corp: 1180/36Kb lim: 4 exec/s: 292 rss: 264Mb 
I  Step #4: #4096	pulse  cov: 6052 ft: 7609 corp: 2319/72Kb lim: 4 exec/s: 256 rss: 267Mb 
I  Step #4: ==37==ERROR: UndefinedBehaviorSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7fc34ee60763 bp 0x7ffee409c820 sp 0x7ffee409c780 T37) 
I  Step #4: ==37==The signal is caused by a READ memory access. 
I  Step #4: ==37==Hint: address points to the zero page. 
I  Step #4:     #0 0x7fc34ee60762 in mozilla::AntiTrackingCommon::StoreUserInteractionFor(nsIPrincipal*) /src/mozilla-central/toolkit/components/antitracking/AntiTrackingCommon.cpp:1139:34 
I  Step #4:     #1 0x7fc34d01e1b3 in mozilla::dom::ContentParent::RecvStoreUserInteractionAsPermission(IPC::Principal const&) /src/mozilla-central/dom/ipc/ContentParent.cpp:5960:3 
I  Step #4:     #2 0x7fc349ce7f84 in mozilla::dom::PContentParent::OnMessageReceived(IPC::Message const&) /work/obj-fuzz/ipc/ipdl/PContentParent.cpp:8149:20 
I  Step #4:     #3 0x7fc34fc439e4 in void mozilla::ipc::FuzzProtocol<mozilla::dom::ContentParent>(mozilla::dom::ContentParent*, unsigned char const*, unsigned long, nsTArray<nsTString<char> > const&) /work/obj-fuzz/dist/include/ProtocolFuzzer.h:117:18 
I  Step #4:     #4 0x7fc34fc43579 in RunContentParentIPCFuzzing(unsigned char const*, unsigned long) /src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp:33:3 
I  Step #4:     #5 0x56490f2ba263 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (out/profile/firefox/firefox+0x100263) 
I  Step #4:     #6 0x56490f2b9a8d in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool*) (out/profile/firefox/firefox+0xffa8d) 
I  Step #4:     #7 0x56490f2bb6e6 in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, fuzzer::fuzzer_allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) (out/profile/firefox/firefox+0x1016e6) 
I  Step #4:     #8 0x56490f2bba2d in fuzzer::Fuzzer::Loop(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, fuzzer::fuzzer_allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) (out/profile/firefox/firefox+0x101a2d) 
I  Step #4:     #9 0x56490f29d5b1 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (out/profile/firefox/firefox+0xe35b1) 
I  Step #4:     #10 0x7fc34f2066a4 in mozilla::FuzzerRunner::Run(int*, char***) /src/mozilla-central/tools/fuzzing/interface/harness/FuzzerRunner.cpp:60:10 
I  Step #4:     #11 0x7fc34f18140c in XREMain::XRE_mainStartup(bool*) /src/mozilla-central/toolkit/xre/nsAppRunner.cpp:3960:35 
I  Step #4:     #12 0x7fc34f18833b in XREMain::XRE_main(int, char**, mozilla::BootstrapConfig const&) /src/mozilla-central/toolkit/xre/nsAppRunner.cpp:4911:12 
I  Step #4:     #13 0x7fc34f1890ec in XRE_main(int, char**, mozilla::BootstrapConfig const&) /src/mozilla-central/toolkit/xre/nsAppRunner.cpp:5018:21 
I  Step #4:     #14 0x56490f262cf4 in do_main(int, char**, char**) (out/profile/firefox/firefox+0xa8cf4) 
I  Step #4:     #15 0x56490f262971 in main (out/profile/firefox/firefox+0xa8971) 
I  Step #4:     #16 0x7fc39390f82f in __libc_start_main (out/profile/./lib/libc.so.6+0x2082f) 
I  Step #4:     #17 0x56490f237868 in _start (out/profile/firefox/firefox+0x7d868) 
I  Step #4:  
I  Step #4: UndefinedBehaviorSanitizer can not provide additional info. 
I  Step #4: SUMMARY: UndefinedBehaviorSanitizer: SEGV /src/mozilla-central/toolkit/components/antitracking/AntiTrackingCommon.cpp:1139:34 in mozilla::AntiTrackingCommon::StoreUserInteractionFor(nsIPrincipal*) 
I  Step #4: ==37==ABORTING 
I  Step #4: MS: 0 ; base unit: 0000000000000000000000000000000000000000 
I  Step #4: 0x8,0x0,0x0,0x0,0xff,0xff,0xff,0x7f,0x15,0x1,0x30,0x0,0x0,0xff,0x0,0x0,0xff,0xd0,0xd0,0xd0,0xd0,0xd0,0x3b,0xd0,0x0,0x0,0x46,0x46,0x46,0x71,0x31,0x30,0x30,0x30,0xf4,0x33,0x34,0x33,0x36,0x38, 
I  Step #4: \x08\x00\x00\x00\xff\xff\xff\x7f\x15\x010\x00\x00\xff\x00\x00\xff\xd0\xd0\xd0\xd0\xd0;\xd0\x00\x00FFFq1000\xf434368 
I  Step #4: artifact_prefix='./'; Test unit written to ./crash-08c02838cf3c3a0757c29af132c1462c2a26eb1d 
I  Step #4: Base64: CAAAAP///38VATAAAP8AAP/Q0NDQ0DvQAABGRkZxMTAwMPQzNDM2OA== 
I  Step #4: du: cannot access '/workspace/out/profile/dumps/./ContentParentIPC.*.profraw': No such file or directory 
I  Step #4: warning: 199 functions have mismatched data 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE7emplaceIJRDiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: Make sure that all profile data to be merged is generated from the same binary. 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiEdeEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE7emplaceIJRiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./SdpParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: [2018-09-25 14:41:18,677 INFO] Finding shared libraries for targets (if any). 
I  Step #4: [2018-09-25 14:41:18,686 INFO] Finished finding shared libraries for targets. 
I  Step #4: warning: 199 functions have mismatched data 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE7emplaceIJRDiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: Make sure that all profile data to be merged is generated from the same binary. 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.15886135739147999814_0.profraw: _ZN7mozilla5MaybeIDiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiEdeEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE7emplaceIJRiEEEvDpOT_: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.4343716965573632615_0.profraw: _ZNK7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: /workspace/out/profile/dumps/./ContentSecurityPolicyParser.4343716965573632615_0.profraw: _ZN7mozilla5MaybeIiE3refEv: Function basic block count change detected (counter mismatch) 
I  Step #4: [2018-09-25 14:51:14,474 INFO] Finding shared libraries for targets (if any). 
I  Step #4: [2018-09-25 14:51:14,495 INFO] Finished finding shared libraries for targets. 
I  Step #4: warning: 257 functions have mismatched data 
I  Step #4: [2018-09-25 14:53:39,828 INFO] Finding shared libraries for targets (if any). 
I  Step #4: [2018-09-25 14:53:39,849 INFO] Finished finding shared libraries for targets. 
I  Step #4: [0;31mwarning: 257 functions have mismatched data 
I  Step #4: [0mwarning: 257 functions have mismatched data 
I  Step #4: [2018-09-25 15:08:45,031 DEBUG] Finished generating per-file code coverage summary. 
I  Step #4: [2018-09-25 15:08:45,032 DEBUG] Generating file view html index file as: "/workspace/out/profile/report/linux/file_view_index.html". 
I  Step #4: [2018-09-25 15:08:46,363 DEBUG] Finished generating file view html index file. 
I  Step #4: [2018-09-25 15:08:46,388 DEBUG] Calculating per-directory coverage summary. 
I  Step #4: [2018-09-25 15:08:47,148 DEBUG] Finished calculating per-directory coverage summary. 
I  Step #4: [2018-09-25 15:08:47,148 DEBUG] Writing per-directory coverage html reports. 
I  Step #4: [2018-09-25 15:09:02,875 DEBUG] Finished writing per-directory coverage html reports. 
I  Step #4: [2018-09-25 15:09:02,875 DEBUG] Generating directory view html index file as: "/workspace/out/profile/report/linux/directory_view_index.html". 
I  Step #4: [2018-09-25 15:09:02,876 DEBUG] Finished generating directory view html index file. 
I  Step #4: [2018-09-25 15:09:02,877 INFO] Index file for html report is generated as: "file:///workspace/out/profile/report/linux/index.html". 
I  Finished Step #4 
```

**Comment by Dor1s** [comment]
Updates on the other targets:

* **dlplibs** fits the disk now, but something else is broken (all coverage dumps are malformed): https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10590
* **libpsl** still the same build failure: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10591
* **strongswan** still the same build failure: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10238

All the others are good now.

**Comment by pdknsk** [comment]
Why is *profile* running with `UndefinedBehaviorSanitizer`? That doesn't seem right because it's build without.

**Comment by pdknsk** [comment]
It turns out sanitizer-less `libFuzzer` generally reports `SIG*` as `UndefinedBehaviorSanitizer`.

I had expected crashes to be filtered out for coverage runs though. I notice target `ContentParentIPC` is missing from the report because of that, but is still listed with percentages for `edge_cov` and `func_cov` in the dashboard.

**Comment by Dor1s** [comment]
Hmm, that's really weird that there are any crashes on the corpus backup, but I have a solution for that, that's fine.

**Comment by pdknsk** [comment]
It appears the reason is that crash being [marked flaky](https://oss-fuzz.com/v2/testcase-detail/5686413949927424) when it was found.
```
oss-fuzz-linux-zone1-host-8rs3-5: Minimize task started.
oss-fuzz-linux-zone1-host-8rs3-5: Minimize task errored out: Testcase appears to be flaky.
```

**Comment by pdknsk** [comment]
I noticed significantly lower coverage reported today for `firefox` compared to the past few days. It turns out target `ContentParentIPC` did a coverage run with only a small subset of its corpus. Please compare counts [yesterday](https://storage.googleapis.com/oss-fuzz-coverage/firefox/reports/20181005/linux/src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp.html) and [today](https://storage.googleapis.com/oss-fuzz-coverage/firefox/reports/20181006/linux/src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp.html). There are no errors or problems in the [build log](https://oss-fuzz-build-logs.storage.googleapis.com/log-6d20c17d-d710-4375-abcb-285e82878113.txt). Either a bug or just a glitch.

**Comment by Dor1s** [comment]
@pdknsk, thanks for spotting that!

The most recent two reports look good again ([Oct 7th](https://storage.googleapis.com/oss-fuzz-coverage/firefox/reports/20181007/linux/src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp.html), [Oct 8th](https://storage.googleapis.com/oss-fuzz-coverage/firefox/reports/20181008/linux/src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp.html)).

The thing is that we're using `-merge=1` instead of `-runs=0` when generating coverage reports (to be crash resistant). [ContentParentIPC.log](https://storage.googleapis.com/oss-fuzz-coverage/firefox/logs/20181006/ContentParentIPC.log) for that "unlucky" day is different from a usual log:

```
Running Fuzzer tests...
INFO: Seed: 1606903083
INFO: Loaded 1 modules   (1388096 inline 8-bit counters): 1388096 [0x7f23c339df58, 0x7f23c34f0d98), 
INFO: Loaded 1 PC tables (1388096 PCs): 1388096 [0x7f23c34f0d98,0x7f23c4a1f198), 
MERGE-OUTER: 15933 files, 0 in the initial corpus
MERGE-OUTER: attempt 1
MERGE-OUTER: attempt 2
MERGE-OUTER: attempt 3
MERGE-OUTER: attempt 4
MERGE-OUTER: attempt 5
MERGE-OUTER: attempt 6
MERGE-OUTER: attempt 7
MERGE-OUTER: succesfull in 7 attempt(s)
MERGE-OUTER: the control file has 3627464 bytes
MERGE-OUTER: consumed 2Mb (262Mb rss) to parse the control file
MERGE-OUTER: 9063 new files with 96822 new features added
```

Usually we have `MERGE-OUTER: attempt 1` only. I suspect that the fuzz target might've had crashes on corpus backup on that day, and thus reading the corpus was not as smooth as it should be, resulting in an unusual coverage numbers.

**Comment by pdknsk** [comment]
> I suspect that the fuzz target might've had crashes on corpus backup on that day, and thus reading the corpus was not as smooth as it should be, resulting in an unusual coverage numbers.

It did report a [bug](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10826) for that target on the same day, so maybe that's related.

**Comment by pdknsk** [comment]
Same problem [today](https://storage.googleapis.com/oss-fuzz-coverage/firefox/reports/20181011/linux/src/mozilla-central/dom/ipc/fuzztest/content_parent_ipc_libfuzz.cpp.html).

I don't know if it's worth the effort, but have you thought about filtering out crashes through sth. similar to a binary search algo? Might work relatively efficiently.

**Comment by Dor1s** [comment]
Hm, we usually don't have crashes when running on a corpus backup. Doing another corpus cleanup (after the one which is already done by corpus pruning task) seems a bit excessive. Do you know if the target is indeed crashing or that's not really the case?

**Comment by pdknsk** [comment]
> Do you know if the target is indeed crashing or that's not really the case?

I don't know.

**Comment by pdknsk** [comment]
It's still happening sporadically. I think this may be related to the general flakiness of `ContentParentIPC` bugs. I'm suspecting some external factor, like `run_minijail` or perhaps even instance differences, which seems unlikely. I'll move the discussion to a separate bug should it still occur after `run_minijail` is removed.

**Comment by Dor1s** [comment]
I think this can be closed now. All failing projects have auto-reported issues in the bug tracker, and I don't see any fundamental problems with the build infrastructure.
